File: frontend/src/model/Collection.ts
import { Item } from "./Item";

export class Collection {

    constructor(
        private _id: number,
        private _title: string,
        private _linkUrl: string,
        private _imageUrl: string,
        private _items: Array<Item>

    ) { }

    public get imageUrl(): string {
        return this._imageUrl;
    }
    public set imageUrl(value: string) {
        this._imageUrl = value;
    }

    public get items(): Array<Item> {
        return this._items;
    }
    public set items(value: Array<Item>) {
        this._items = value;
    }
    public get linkUrl(): string {
        return this._linkUrl;
    }
    public set linkUrl(value: string) {
        this._linkUrl = value;
    }
    public get title(): string {
        return this._title;
    }
    public set title(value: string) {
        this._title = value;
    }
    public get id(): number {
        return this._id;
    }
    public set id(value: number) {
        this._id = value;
    }
}

File: backend/src/main/java/com/clothing/webservices/demo/model/Collection.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.clothing.webservices.demo.model;

import java.io.Serializable;
import java.util.List;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.OneToMany;
import javax.persistence.Table;

/**
 *
 * @author Chanh Thanh
 */
@Entity
@Table(name = "Collection")
public class Collection implements Serializable {

    @Id
    @GeneratedValue
    private int id;

    @Column(name = "imageUrl")
    private String imageUrl;

    @Column(name = "title")
    private String title;

    @Column(name = "linkUrl")
    private String linkUrl;
    
    @OneToMany(mappedBy = "collection")
    private List<Item> items;

    protected Collection() {
    }

    public Collection(int id, String imageUrl, String title, String linkUrl, List<Item> items) {
        this.id = id;
        this.imageUrl = imageUrl;
        this.title = title;
        this.linkUrl = linkUrl;
        this.items = items;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getImageUrl() {
        return imageUrl;
    }

    public void setImageUrl(String imageUrl) {
        this.imageUrl = imageUrl;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getLinkUrl() {
        return linkUrl;
    }

    public void setLinkUrl(String linkUrl) {
        this.linkUrl = linkUrl;
    }

    public List<Item> getItems() {
        return items;
    }

    public void setItems(List<Item> items) {
        this.items = items;
    }

}
